app/services/add_users.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class AddUsers 
+  def initialize(course:, usernames:, current_user:)
+    @course = course
+    @usernames = usernames
+    @current_user = current_user
+  end
+
+  def add_all_at_once
+    usernames_list = @usernames.lines.map(&:strip)
```

Comment:
I think doing this in the controller makes a little more sense. That way, the interface to this service expects an array of usernames (rather than a multiline string), which will be more intuitive and easier to reuse.

The Rails convention is for controllers to handle user input, and service objects to carry out business logic.